public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/response_parser.ts

```diff
@@ -147,6 +147,27 @@ export default class ResponseParser {
   static dateTimeToEpoch(dateTimeValue: any) {
     return dateTime(dateTimeValue).valueOf();
   }
+
+  static parseSubscriptions(result: any): Array<{ text: string; value: string }> {
+    const list: Array<{ text: string; value: string }> = [];
+
+    if (!result) {
+      return list;
+    }
+
+    const valueFieldName = 'subscriptionId';
+    const textFieldName = 'displayName';
+    for (let i = 0; i < result.data.value.length; i++) {
+      if (!find(list, ['value', get(result.data.value[i], valueFieldName)])) {
+        list.push({
+          text: `${get(result.data.value[i], textFieldName)}`,
```

Comment:
I think this could be a bit easier to understand if it used a for-of loop. And does just the Array.prototype.find function work? (i try to avoid using lodash where we have native methods)

Reply:
Actually, I see this was just copied from the existing function in azure_monitor/response_parser.

Lets just leave this as is, and I'll capture a task for us to refactor all of this anyway. It's not great code and there's a lot of duplication between metrics and logs.